starlark/eval_test.go

```diff
@@ -1109,3 +1109,44 @@ f(1)
 		t.Errorf("env() returned %s, want %s", got, want)
 	}
 }
+
+func TestUnpackArgsOptionalInference(t *testing.T) {
+	// success
+	kwargs := []starlark.Tuple{
+		{starlark.String("x"), starlark.MakeInt(1)},
+		{starlark.String("y"), starlark.MakeInt(2)},
+	}
+	var x, y, z int
+	if err := starlark.UnpackArgs("unpack", nil, kwargs,
+		"x", &x, "y?", &y, "z", &z); err != nil {
+		t.Errorf("UnpackArgs failed: %v", err)
+	}
+	if x != 1 {
```

Comment:
You can achieve higher vertical density (and, in this case, readability) in both the code and the test log by combining all three in one line:

got := fmt.Sprint("x=%d y=%d z=%d", x, y, z)
want := "x=1 y=2 z=0"
...assert...